vector_quantize_pytorch/finite_scalar_quantization.py

```diff
@@ -0,0 +1,65 @@
+"""
+Finite Scalar Quantization: VQ-VAE Made Simple - https://arxiv.org/abs/2309.15505
+Code adapted from Jax version in Appendix A.1
+"""
+
+import torch
+import torch.nn as nn
+
+
+def round_ste(z: torch.Tensor) -> torch.Tensor:
+    """Round with straight through gradients."""
+    zhat = z.round()
+    return z + (zhat - z).detach()
+
+
+class FSQ(nn.Module):
```

Comment:
Thanks so much for porting this! Do you mind if we link this repo in the next version and our own public code release?

Reply:
LMK if you are also planning to update the README and I can send some figs.

Reply:
please do! and i believe @lucidrains and @sekstini will appreciate it

Reply:
Yup, go ahead 👍

Reply:
@fab-jul

> LMK if you are also planning to update the README and I can send some figs.

That would be great 🙏